extensions/amp-ooyala-player/0.1/amp-ooyala-player.js

```diff
@@ -103,31 +120,68 @@ class AmpOoyalaPlayer extends AMP.BaseElement {
 
   /** @override */
   pauseCallback() {
-    if (this.iframe_) {
+    // Only send pauseVideo command if the player is playing. Otherwise
+    // The player breaks if the user haven't played the video yet specially
+    // on mobile.
+    if (this.iframe_ && this.iframe_.contentWindow &&
+        this.playerState_ == PlayerStates.PLAYING) {
       this.pause();
     }
   }
 
+  /** @private */
+  handleOoyalaMessages_(event) {
+    if (event.origin != 'https://player.ooyala.com' ||
+        event.source != this.iframe_.contentWindow) {
+      return;
+    }
+    if (!event.data ||
```

Comment:
@mityaha These are very YouTube specific, I don't assume ooyala iframe sends the same messages as YouTube. Does it?